### PR TITLE
ddl/ddl_test: migrate test-infra to testify

### LINKED ISSUE #28628: migrate test-infra to testify for `ddl/attributes_sql_test.go`

### PR DESCRIPTION
**Pull request by dengqee**
### What problem does this PR solve?

Issue Number: close #28628 

### Release note

```release-note
None
```

### CODE REVIEW

File: testkit/testkit.go [hunk 1]
```diff
@@ -234,3 +236,28 @@ func WithPruneMode(tk *TestKit, mode variable.PartitionPruneMode, f func()) {
 	tk.MustExec("set global tidb_partition_prune_mode=`" + string(mode) + "`")
 	f()
 }
+
+// MockGC is used to make GC work in the test environment.
+func MockGC(tk *TestKit) (string, string, string, func()) {
```
Comment (zimulala) [inline testkit/testkit.go:241]: It seems that this function doesn't be used. Could we remove it?

Comment (dengqee) [reply testkit/testkit.go:247]: It was called by attributes_sql_test.go for many times.
like: https://github.com/pingcap/tidb/blob/6495a5a116a016a3e077d181b8c8ad81f76ac31b/ddl/attributes_sql_test.go#L252

**Review by hawkingrei — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
@dengqee thanks for your contribution!

File: testkit/testkit.go (hunk 1 above)
Comment (hawkingrei) [reply testkit/testkit.go:271]: It will lead to cycle import when ddl_test use ```testkit```. so I will remove it at #32058.

**Pull request merged**